src/test/scala/mesosphere/marathon/core/launchqueue/impl/ReviveOffersActorTest.scala

```diff
@@ -0,0 +1,174 @@
+      Mockito.verify(f.driver, f.invocationTimeout).suppressOffers()
+
+      f.verifyNoMoreInteractions()
+    }
+
+    "revive and suppress upon instance updates" in {
+      val f = new Fixture()
+
+      Given("no initial instances")
+      f.instanceTracker.instancesBySpec()(any[ExecutionContext]) returns Future.successful(InstanceTracker.InstancesBySpec.empty)
+
+      When("the actor initializes")
+      f.actorRef.start()
+
+      Then("it will suppress offers")
+      Mockito.verify(f.instanceTracker).instancesBySpec()(any[ExecutionContext])
+      Mockito.verify(f.driver, f.invocationTimeout).suppressOffers()
+
+      When("the actor gets notified of a new Scheduled instance")
+      val instance1 = Instance.scheduled(f.app)
+      val instance1ChangeEvent = InstanceChangedEventsGenerator.updatedCondition(instance1)
+      system.eventStream.publish(instance1ChangeEvent)
+
+      Then("reviveOffers is called")
+      Mockito.verify(f.driver, f.invocationTimeout).reviveOffers()
+
+      When("the actor gets notified of another Scheduled instance")
+      val instance2 = Instance.scheduled(f.app)
+      val instance2ChangeEvent = InstanceChangedEventsGenerator.updatedCondition(instance2)
+      system.eventStream.publish(instance2ChangeEvent)
+
+      Then("reviveOffers is called again, since we might have declined offers meanwhile")
+      Mockito.verify(f.driver, times(2)).reviveOffers()
+
+      When("the actor gets notified of the first instance becoming Staging")
+      val instance1Staging = TestInstanceBuilder.newBuilderWithInstanceId(instance1.instanceId).addTaskStaged().getInstance()
+      val instance1StagingEvent = InstanceChangedEventsGenerator.updatedCondition(instance1Staging)
+      system.eventStream.publish(instance1StagingEvent)
+
+      And("the actor gets notified of the second instance becoming Gone")
+      val instance2Gone = TestInstanceBuilder.newBuilderWithInstanceId(instance2.instanceId).addTaskGone().getInstance()
+      val instance2GoneEvent = InstanceChangedEventsGenerator.updatedCondition(instance2Gone)
+      system.eventStream.publish(instance2GoneEvent)
+
+      Then("suppress is called again")
+      Mockito.verify(f.driver, times(2)).suppressOffers()
+
+      f.verifyNoMoreInteractions()
+    }
+  }
+
+  class Fixture() {
+    implicit val mat: ActorMaterializer = ActorMaterializer()
+    val conf: ReviveOffersConfig = {
+      new ReviveOffersConfig {
+        verify()
+      }
+    }
+    val driver: SchedulerDriver = mock[SchedulerDriver]
+    val driverHolder: MarathonSchedulerDriverHolder = {
+      val holder = new MarathonSchedulerDriverHolder
+      holder.driver = Some(driver)
+      holder
+    }
+    val instanceTracker: InstanceTracker = mock[InstanceTracker]
+    instanceTracker.instanceUpdates returns Source.empty // TODO: needed?
+
+    val metrics: Metrics = DummyMetrics
+
+    lazy val actorRef: TestActorRef[ReviveOffersActor] = TestActorRef[ReviveOffersActor](
```

Comment:
Why not make the logic a simple trait or class such as in `ExpungeOverdueLostTasksActorLogic`. This wy we don't need to test the actor but a simple class which is much faster and simpler. No racy tests!